File: app/src/main/java/com/example/webviewapp/MainActivity.java

package com.example.webviewapp;

import android.os.Bundle;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.widget.Button;
import android.widget.EditText;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        WebView webView = findViewById(R.id.webview);
        EditText inputUrl = findViewById(R.id.input_url);
        EditText inputUa = findViewById(R.id.input_ua);
        Button goButton = findViewById(R.id.go_button);

        WebSettings webSettings = webView.getSettings();
        webSettings.setJavaScriptEnabled(true);
        webSettings.setMediaPlaybackRequiresUserGesture(false);
        webSettings.setDomStorageEnabled(true);
        webSettings.setAllowContentAccess(true);
        webSettings.setAllowFileAccess(true);
        webView.clearCache(true);

        goButton.setOnClickListener(v -> {
            String url = inputUrl.getText().toString();
            String ua = inputUa.getText().toString();
            webSettings.setUserAgentString(ua);
            webView.loadUrl(url);
        });
    }
}

File: app/src/main/java/com/example/customwebview/MainActivity.kt
package com.example.customwebview

import android.os.Bundle
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private lateinit var webView: WebView
    private lateinit var urlInput: EditText
    private lateinit var userAgentInput: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        webView = findViewById(R.id.webView)
        urlInput = findViewById(R.id.urlInput)
        userAgentInput = findViewById(R.id.userAgentInput)

        setupWebView()

        findViewById<Button>(R.id.loadButton).setOnClickListener {
            loadUrl()
        }

        findViewById<Button>(R.id.setUserAgentButton).setOnClickListener {
            setUserAgent()
        }
    }

    private fun setupWebView() {
        val webSettings = webView.settings
        
        // Основні налаштування
        webSettings.javaScriptEnabled = true
        webSettings.domStorageEnabled = true
        
        // Вимкнення WebRTC
        webSettings.mediaPlaybackRequiresUserGesture = true
        webSettings.mixedContentMode = WebSettings.MIXED_CONTENT_NEVER_ALLOW
        
        webView.webViewClient = WebViewClient()
        urlInput.setText("https://google.com")
        loadUrl()
    }

    private fun loadUrl() {
        var url = urlInput.text.toString().trim()
        if (!url.startsWith("http://") && !url.startsWith("https://")) {
            url = "https://$url"
        }
        webView.loadUrl(url)
    }

    private fun setUserAgent() {
        val userAgent = userAgentInput.text.toString().trim()
        if (userAgent.isNotEmpty()) {
            webView.settings.userAgentString = userAgent
        }
    }

    override fun onBackPressed() {
        if (webView.canGoBack()) {
            webView.goBack()
        } else {
            super.onBackPressed()
        }
    }
}